openeogeotrellis/GeotrellisImageCollection.py

```diff
@@ -514,131 +519,102 @@ def timeseries(self, x, y, srs="EPSG:4326") -> Dict:
+                stats = self._compute_stats_geotrellis().compute_histograms_time_series_from_datacube(
+                    scala_data_cube, polygons, from_date.isoformat(), to_date.isoformat(), self._band_index
+                )
             elif func == 'sd':
-                implementation = self._compute_stats_geotrellis().compute_sd_time_series_from_datacube
-            else:
-                raise ValueError(func)
-
-            stats = implementation(
-                scala_data_cube,
-                polygon_wkts,
-                polygons_srs,
-                from_date.isoformat(),
-                to_date.isoformat(),
-                self._band_index
-            )
-
-            return AggregatePolygonResult(
-                timeseries=self._as_python(stats),
-                regions=regions if multiple_geometries else GeometryCollection([regions]),
-            )
-        elif func == 'median':
-            highest_level = self.pyramid.levels[self.pyramid.max_zoom]
-            layer_metadata = highest_level.layer_metadata
-
-            scala_data_cube = highest_level.srdd.rdd()
-
-            from_date = insert_timezone(layer_metadata.bounds.minKey.instant)
-            to_date = insert_timezone(layer_metadata.bounds.maxKey.instant)
-
-            if from_vector_file:
-                stats = self._compute_stats_geotrellis().compute_median_time_series_from_datacube(
-                    scala_data_cube,
-                    regions,
-                    from_date.isoformat(),
-                    to_date.isoformat(),
-                    self._band_index
+                stats = self._compute_stats_geotrellis().compute_sd_time_series_from_datacube(
+                    scala_data_cube, polygons, from_date.isoformat(), to_date.isoformat(), self._band_index
                 )
-            else:
-                polygon_wkts = [str(ob) for ob in regions] if multiple_geometries else [str(regions)]
-                polygons_srs = 'EPSG:4326'
-
+            elif func == 'median':
                 stats = self._compute_stats_geotrellis().compute_median_time_series_from_datacube(
-                    scala_data_cube,
-                    polygon_wkts,
-                    polygons_srs,
-                    from_date.isoformat(),
-                    to_date.isoformat(),
-                    self._band_index
+                    scala_data_cube, polygons, from_date.isoformat(), to_date.isoformat(), self._band_index
                 )
+            else:
+                raise ValueError(func)
 
             return AggregatePolygonResult(
                 timeseries=self._as_python(stats),
+                # TODO: regions can also be a string (path to vector file) instead of geometry object
                 regions=regions if multiple_geometries else GeometryCollection([regions]),
             )
-        else:  # defaults to mean, historically
+        elif func == "mean":  # defaults to mean, historically
```

Comment:
if it no longer defaults to "mean" I'd get rid of the comment

Reply:
True.
I don't think we have to support "default to mean" anymore